src/textual/app.py

```diff
@@ -1321,6 +1337,14 @@ def push_screen(self, screen: Screen | str) -> AwaitMount:
             screen: A Screen instance or the name of an installed screen.
 
         """
+        if not isinstance(screen, (Screen, str)):
+            raise TypeError(
+                f"push_screen requires a Screen instance or str; not {screen!r}"
+            )
+
+        if self._screen_stack:
+            self.screen.post_message(events.ScreenSuspend())
+            self.screen.refresh()
```

Comment:
I don't understand why these two method calls on `self.screen` are guarded by the `if`.
Is this to check if the currently active screen is _not_ the app default screen?

Reply:
This may be called when there are no screens, and hence `self.screen` raises an exception.